palette/script.js

```diff
@@ -0,0 +1,163 @@
+                drawRect(x, y, deltaX, deltaY, curColor);
+            }
+        } else {
+            for (let y = y0; y >= y1; y -= deltaY) {
+                const x = f(y);
+                drawRect(x, y, deltaX, deltaY, curColor);
+            }
+        }
+    }
+}
+
+function ColorPicker(event, arr) {
+    document.getElementById('prev_color').style.background = CurrentColor();
+    localStorage.setItem('prev_color', CurrentColor());
+    document.getElementById('current_color').style.background = arr[Math.floor(event.offsetY / 128)][Math.floor(event.offsetX / 128)];
+    localStorage.setItem('current_color', CurrentColor());
+}
+
+window.onload = () => {
+    let arr = localStorage.getItem('canvas') ? JSON.parse(localStorage.getItem('canvas')) : smallImg;
+    drawArray(4, arr);
+    const canvas = document.getElementsByTagName('canvas')[0];
+    canvas.addEventListener('click', (event) => {
+        switch (document.querySelector('.tools-block__tools_item_active').children[1].innerHTML) {
+        case 'Pencil':
+            Pencil(event, arr);
+            drawArray(4, arr);
+            break;
+        case 'Choose color':
+            ColorPicker(event, arr);
+            break;
+        case 'Paint bucket':
+            arr = Filling(arr);
+            drawArray(4, arr);
+            break;
+        default:
+            break;
+        }
+
+        localStorage.setItem('canvas', JSON.stringify(arr));
+    });
+
+    canvas.addEventListener('mousemove', (event) => {
+        if (event.which === 1 && document.getElementsByClassName('tools-block__tools_item_active')[0].children[1].innerHTML === 'Pencil') {
+            arr[Math.floor(event.offsetY / 128)][Math.floor(event.offsetX / 128)] = CurrentColor();
+            drawArray(4, arr);
+            localStorage.setItem('canvas', JSON.stringify(arr));
+        }
+    });
+};
+
+Array.from(document.getElementsByClassName('tools-block__tools_item')).forEach((element) => {
+    element.addEventListener('click', () => {
+        document.getElementsByClassName('tools-block__tools_item_active')[0].classList.remove('tools-block__tools_item_active');
+
+        element.classList.add('tools-block__tools_item_active');
+    });
+});
+
+Array.from(document.getElementsByClassName('change')).forEach((element) => {
+    element.addEventListener('click', () => {
+        if (element.children[0].id !== 'prev_color') {
+            document.getElementById('prev_color').style.background = CurrentColor();
+            localStorage.setItem('prev_color', CurrentColor());
+            document.getElementById('current_color').style.background = window.getComputedStyle(element.children[0], null).getPropertyValue('background-color');
+            localStorage.setItem('current_color', CurrentColor());
+        } else {
+            document.getElementById('current_color').style.background = window.getComputedStyle(element.children[0], null).getPropertyValue('background-color');
+        }
+    });
```

Comment:
try to find all elements outside the listener 
...or even function
...or even at the beginning of your code

Reply:
ok, fixed